crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -97,30 +116,31 @@ impl Default for StarknetState {
 impl StarknetState {
     pub fn new(defaulter: StarknetDefaulter) -> Self {
         Self {
-            state: CachedState::new(
-                DictState::new(defaulter),
-                GlobalContractCache::new(GLOBAL_CONTRACT_CACHE_SIZE_FOR_TEST),
-            ),
-            rpc_contract_classes: Default::default(),
+            state: CachedState::new(DictState::new(defaulter), default_global_contract_cache()),
+            rpc_contract_classes: Arc::new(RwLock::new(CommittedClassStorage::default())),
             historic_state: Default::default(),
         }
     }
 
     pub fn clone_rpc_contract_classes(&self) -> CommittedClassStorage {
-        self.rpc_contract_classes.clone()
+        self.rpc_contract_classes.read().clone()
     }
 
     /// Commits and returns the state difference accumulated since the previous (historic) state.
-    pub(crate) fn commit_with_diff(&mut self) -> DevnetResult<StateDiff> {
-        let diff = StateDiff::generate(&mut self.state, &mut self.rpc_contract_classes)?;
+    /// Classes are NOT committed; see `Self::commit_classes`.
+    pub(crate) fn commit_diff(&mut self) -> DevnetResult<StateDiff> {
+        let new_classes = self.rpc_contract_classes.read().staging.clone();
+
+        let diff = StateDiff::generate(&mut self.state, new_classes)?;
         let new_historic = self.expand_historic(diff.clone())?;
-        self.state = CachedState::new(
-            new_historic.clone(),
-            GlobalContractCache::new(GLOBAL_CONTRACT_CACHE_SIZE_FOR_TEST),
-        );
+        self.state = CachedState::new(new_historic.clone(), default_global_contract_cache());
         Ok(diff)
     }
 
+    pub(crate) fn commit_classes(&mut self, block_number: u64) {
```

Comment:
Here and in other places why are we using block numbers instead of block hashes? Won't this threaten the abortion block functionality? Tests with abortion block functionality are missing.

Reply:
I'll have to think more about the impact on block abortion

Reply:
Properly handling block abortion in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/499/commits/ba6f944073fa50dee1c369396c854ffaeb168b1f